### ISSUE TITLE
log-likelihood from eval_model for seq2seq

### ISSUE THREAD

**Issue by sarapapi** [ThilinaRajapakse/simpletransformers#809]
Hi, 
I'm trying to obtain the log-likelihood from T5 during the eval_model step, I think that it is useful for computing perplexity [like in HF doc](https://huggingface.co/transformers/perplexity.html) or maybe it is useful itself. 
Is there any workaround for it? It might be useful to add it as an output for seq2seq models instead of an only loss.

**Comment by ThilinaRajapakse** [comment]
I think we could just return the model outputs from the `evaluate()` and `eval_model()` methods (since this is how all other models behave). The user can then use the outputs to calculate perplexity (or anything else for that matter). This applies to both T5 and Seq2Seq.

What do you think?

**Comment by sarapapi** [comment]
> I think we could just return the model outputs from the `evaluate()` and `eval_model()` methods (since this is how all other models behave). The user can then use the outputs to calculate perplexity (or anything else for that matter). This applies to both T5 and Seq2Seq.
> 
> What do you think?

Overall perplexity, for example, can be computed as `exp(eval_loss)` that the `evaluate()` returns and it is can be done trivially.
If, instead, I'm interested in Sentence perplexity the only way, without modifying the library, is to feed each sentence to `evaluate()` separately, collecting the loss and then compute the exponential. Am I right?
But what if I'm interested in something else? Like, for example, the log-likelihood? I have to modify `evaluate()` to return all the outputs [as in docs](https://huggingface.co/transformers/model_doc/t5.html#t5forconditionalgeneration), like "logits" but I'm not sure about how to obtain the sentence probability itself. Maybe there is another way, much more simple, I don't know. What do you think about it?

**Comment by ThilinaRajapakse** [comment]
I'm not sure about sentence perplexity, but I think you are right.

You can just use `torch.nn.functional.F.log_softmax` on the logits to get the probabilities unless I am missing something.

**Comment by sarapapi** [comment]
> I'm not sure about sentence perplexity, but I think you are right.
> 
> You can just use `torch.nn.functional.F.log_softmax` on the logits to get the probabilities unless I am missing something.

I don't know if applying the `torch.nn.functional.F.log_softmax` to `logits` output returns the sentence probability, maybe it returns the probability of each word?

**Comment by ThilinaRajapakse** [comment]
It will return the probability for each token, yes. Looks like I need to look into sentence probability/perplexity stuff. I'm not familiar with those.

**Comment by sarapapi** [comment]
> It will return the probability for each token, yes. Looks like I need to look into sentence probability/perplexity stuff. I'm not familiar with those.

Yes, exactly but I want to obtain the sentence probability (as the log-likelihood computed using the original T5 library by Google), so I have to multiply the word probabilities but in tensor format, I think

**Comment by sarapapi** [comment]
I've applied the log_softmax to the first sentence logits output and the results is a tensorf of torch.Size([1, 512, 32100]
`import torch.nn.functional as F
log_softmax = F.log_softmax(logits,dim=1)`
Is the code that I have used, very simple
Then, how can I go on? I have to select the argmax of each token (from 1 to 512) to obtain the probability and then add them (since I used LOG softmax)?
